src/bytecode/astgen.rs

```diff
@@ -284,9 +289,57 @@ impl<'a, 'ast> AstBytecodeGen<'a, 'ast> {
         dest
     }
 
+    fn visit_expr_assert(&mut self, expr: &ExprCallType, _dest: DataDest) {
+        let lbl_assert = self.gen.create_label();
+
+        let assert_reg = self.visit_expr(&*expr.args[0], DataDest::Alloc);
+
+        self.gen.emit_jump_if_true(assert_reg, lbl_assert);
+
+        let cls_id = self.vm.vips.error_class;
+        let cls = self.vm.classes.idx(cls_id);
+        let cls = cls.read();
+
+        let fct_id = cls.constructor.expect("constructor is missing");
+        let fct = self.vm.fcts.idx(fct_id);
+        let fct = fct.read();
+
+        let arg_types = fct
+            .params_with_self()
+            .iter()
+            .map(|&arg| {
+                self.specialize_type_for_call(
+                    &CallType::CtorNew(cls_id, fct_id, TypeParams::Empty),
+                    arg,
+                )
+                .into()
+            })
+            .collect::<Vec<BytecodeType>>();
+        let num_args = arg_types.len();
+
+        let start_reg = self.gen.add_register_chain(&arg_types);
+        let cls_id = specialize_class_id_params(self.vm, cls_id, &TypeParams::Empty);
+        self.gen.emit_new_object(start_reg, cls_id);
+        let error_string_reg = start_reg.offset(1);
+        let error_string_index = self.gen.add_string_const_pool("assert failed".to_string());
+        self.gen
+            .emit_const_string(error_string_reg, error_string_index);
+
+        self.gen
+            .emit_invoke_direct_void(fct_id, start_reg, num_args);
+
+        self.gen.emit_throw(start_reg);
+
+        self.gen.bind_label(lbl_assert);
+    }
+
     fn visit_expr_call(&mut self, expr: &ExprCallType, dest: DataDest) -> Register {
-        if let Some(_intrinsic) = self.get_intrinsic(expr.id) {
-            unimplemented!()
+        if let Some(intrinsic) = self.get_intrinsic(expr.id) {
+            match intrinsic {
+                Intrinsic::Assert => self.visit_expr_assert(expr, dest),
+                _ => unimplemented!(),
+            }
+            return Register::invalid();
```

Comment:
If I unterstand this correctly, assert is not a call with a return value, but (more or less) a statement with side effects. Therefore `Register::invalid()` should be correct.